src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlStringMethodTranslator.cs

```diff
@@ -189,6 +192,19 @@ public SqlExpression Translate(SqlExpression instance, MethodInfo method, IReadO
                         },
                         typeof(int)),
                     _sqlExpressionFactory.Constant(0));
+
+                if (pattern is SqlConstantExpression constantPattern)
+                {
+                    return (string)constantPattern.Value == string.Empty
+                        ? (SqlExpression)_sqlExpressionFactory.Constant(true)
+                        : strposCheck;
+                }
+
+                return _sqlExpressionFactory.OrElse(
```

Comment:
And check the pattern for null here:

```csharp
var pattern = arguments[0];
var constantPattern = pattern as SqlConstantExpression;
if (constantPattern != null && (string)constantPattern.Value == string.Empty)
    return _sqlExpressionFactory.Constant(true);

// ...

return constantPattern is null
    ? _sqlExpressionFactory.OrElse(
        _sqlExpressionFactory.Equal(
            pattern,
            _sqlExpressionFactory.Constant(string.Empty, stringTypeMapping)),
        strposCheck)
    : strposCheck;
```

Reply:
Right, this checks constantPattern for null twice... I think it's cleaner/more understandable as-is, where you have one block handling the constant case and another the non-constance.

Reply:
Without double check this introduces a few more useless allocations, while a null check is very cheap action and it won't hurt performance. If you think that constant patterns isn't a common case, feel free to leave it as is.